internal/controller/controller_utils_test.go

```diff
@@ -0,0 +1,357 @@
+						"leaderworkerset.sigs.k8s.io/group-index":  "1",
+						"leaderworkerset.sigs.k8s.io/group-key":    "test-key",
+					},
+				},
+			},
+			lws: testutils.BuildBasicLeaderWorkerSet("test-sample", "default").
+				Replica(1).
+				WorkerTemplateSpec(testutils.MakeWorkerPodSpec()).
+				Size(1).Obj(),
+			wantStatefulSetConfig: &appsapplyv1.StatefulSetApplyConfiguration{
+				TypeMetaApplyConfiguration: metaapplyv1.TypeMetaApplyConfiguration{
+					Kind:       ptr.To[string]("StatefulSet"),
+					APIVersion: ptr.To[string]("apps/v1"),
+				},
+				ObjectMetaApplyConfiguration: &metaapplyv1.ObjectMetaApplyConfiguration{
+					Name:      ptr.To[string]("test-sample"),
+					Namespace: ptr.To[string]("default"),
+					Labels: map[string]string{
+						"leaderworkerset.sigs.k8s.io/name":        "test-sample",
+						"leaderworkerset.sigs.k8s.io/group-index": "1",
+						"leaderworkerset.sigs.k8s.io/group-key":   "test-key",
+					},
+				},
+				Spec: &appsapplyv1.StatefulSetSpecApplyConfiguration{
+					Replicas: ptr.To[int32](0),
+					Selector: &metaapplyv1.LabelSelectorApplyConfiguration{
+						MatchLabels: map[string]string{
+							"leaderworkerset.sigs.k8s.io/name":        "test-sample",
+							"leaderworkerset.sigs.k8s.io/group-index": "1",
+							"leaderworkerset.sigs.k8s.io/group-key":   "test-key",
+						},
+					},
+					Template: &coreapplyv1.PodTemplateSpecApplyConfiguration{
+						ObjectMetaApplyConfiguration: &metaapplyv1.ObjectMetaApplyConfiguration{
+							Labels: map[string]string{
+								"leaderworkerset.sigs.k8s.io/name":        "test-sample",
+								"leaderworkerset.sigs.k8s.io/group-index": "1",
+								"leaderworkerset.sigs.k8s.io/group-key":   "test-key",
+							},
+							Annotations: map[string]string{
+								"leaderworkerset.sigs.k8s.io/size":        "1",
+								"leaderworkerset.sigs.k8s.io/leader-name": "test-sample",
+							},
+						},
+						Spec: &coreapplyv1.PodSpecApplyConfiguration{
+							Containers: []coreapplyv1.ContainerApplyConfiguration{
+								{
+									Name:      ptr.To[string]("leader"),
+									Image:     ptr.To[string]("nginx:1.14.2"),
+									Ports:     []coreapplyv1.ContainerPortApplyConfiguration{{ContainerPort: ptr.To[int32](8080), Protocol: ptr.To[corev1.Protocol](corev1.ProtocolTCP)}},
+									Resources: &coreapplyv1.ResourceRequirementsApplyConfiguration{},
+								},
+							},
+						},
+					},
+					Ordinals:            &appsapplyv1.StatefulSetOrdinalsApplyConfiguration{Start: ptr.To[int32](1)},
+					ServiceName:         ptr.To[string]("test-sample"),
+					PodManagementPolicy: ptr.To[appsv1.PodManagementPolicyType](appsv1.ParallelPodManagement),
+				},
+			},
+		},
+		{
+			name: "1 replica, size 2, exclusive placement enabled",
+			pod: &corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Name:      "test-sample",
+					Namespace: "default",
+					Labels: map[string]string{
+						"leaderworkerset.sigs.k8s.io/worker-index": "0",
```

Comment:
why is this not showing in wantStatefulSetConfig?

Reply:
The function never adds that label to the statefulsetconfig. I can remove the label from the input if needed